### PR TITLE
Minigame/obstacle jumper

### PR DESCRIPTION
**Pull request by Krischan-Klug**

### CODE REVIEW

File: pages/[id]/minigames/obstacle-jumper/index.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  }, []);
+
+  if (!canvasRef.current) {
+    return null;
+  }
+
+  return (
+    <>
+      <canvas ref={canvasRef}></canvas>
+    </>
```
Comment (onemanwenttomow) [inline pages/[id]/minigames/obstacle-jumper/index.js:180]: No need for a React Fragment. 

```suggestion
      <canvas ref={canvasRef}></canvas>
```

File: pages/[id]/minigames/obstacle-jumper/index.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+      return;
+    }
+
+    let score = 0;
+
+    //Scene Management
+
+    k.scene("start", () => {
+      const start = k.add([
+        k.text("Press to Start"),
```
Comment (onemanwenttomow) [inline pages/[id]/minigames/obstacle-jumper/index.js:31]: It wasn't clear to me how to start the game.

```suggestion
        k.text("Press Space to Start"),
```

File: pages/pet-detail-page/[id].js [hunk 3]
```diff
@@ -345,6 +345,11 @@ export default function PetDetailPage({
             <MoneyColored cost={200} money={userStats.money} /> <MoneyImage />
           </StyledButton>
         )}
+        <StyledButton
+          onClick={() => router.push(`/${id}/minigames/obstacle-jumper`)}
+        >
+          Obstacle Jumper
+        </StyledButton>
```
Comment (onemanwenttomow) [inline pages/pet-detail-page/[id].js:352]: Is there a reason why this is not a link?

File: pages/[id]/minigames/obstacle-jumper/index.js [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        k.text(moneyToAdd + " Coins"),
+        k.pos(k.center().x, k.center().y + 150),
+        k.anchor("center"),
+      ]);
+
+      function endGame() {
+        onAddMoney(moneyToAdd);
+
+        //Router wont work here, reload is needed!
+        window.location.href = `/pet-detail-page/${id}`;
```
Comment (onemanwenttomow) [inline pages/[id]/minigames/obstacle-jumper/index.js:158]: are you sure this doesnt work?

``` js
router.push(`/pet-detail-page/${id}`);
```

**Review by onemanwenttomow — CHANGES_REQUESTED** [review]
Really cool feature 🎉. I have been wanting to play with kaboom for a while so it's really nice to see it being used. 

I added a couple of small thoughts and suggestions.

File: pages/pet-detail-page/[id].js (hunk 3 above)
Comment (Krischan-Klug) [reply pages/pet-detail-page/[id].js:352]: will be a popup in future, that show all minigames

File: pages/[id]/minigames/obstacle-jumper/index.js (hunk 2 above)
Comment (Krischan-Klug) [reply pages/[id]/minigames/obstacle-jumper/index.js:31]: fixed !

File: pages/[id]/minigames/obstacle-jumper/index.js (hunk 1 above)
Comment (Krischan-Klug) [reply pages/[id]/minigames/obstacle-jumper/index.js:180]: fixed !

**Pull request merged**